lib/Sema/TypeCheckType.h

```diff
@@ -455,16 +455,21 @@ class TypeResolution {
   /// \param decl The declaration that the resulting bound generic type
   /// shall reference.
   /// \param parentTy The parent type.
-  /// \param loc The source location for diagnostic reporting.
   /// \param genericArgs The list of generic arguments to apply.
+  /// \param module The parent module of the application context.
+  /// \param loc The source location for diagnostic reporting.
+  /// \param genericEnv The generic environment to be used for requirement
+  /// checking.
   ///
   /// \returns A BoundGenericType bound to the given arguments, or null on
   /// error.
   ///
   /// \see applyGenericArguments
-  Type applyUnboundGenericArguments(GenericTypeDecl *decl, Type parentTy,
-                                    SourceLoc loc,
-                                    ArrayRef<Type> genericArgs) const;
+  static Type
+  applyUnboundGenericArguments(GenericTypeDecl *decl, Type parentTy,
```

Comment:
Would it be cleaner if it was an instance member?

Reply:
On second thought, I have to object on this. It does seem to be visually cleaner the instance member way, but the fact that we sometimes pass a contextual parent type to `applyUnboundGenericArguments` just doesn't play well together with a type resolution stage. It seems to be more about whether we want to check requirements or not, rather than which stage we're on. Looking forward to further thoughts.